pyannote/audio/core/pipeline.py

```diff
@@ -187,6 +182,133 @@ def from_pretrained(
+        tags: Optional[List[str]] = None,
+        cache_dir: Union[Path, Text] = CACHE_DIR,
+        config_yaml_path: str = "pyannote/speaker-diarization-3.1",
+    ):
+        """
+        Upload the pyannote pipeline config file to the 🤗 Model Hub.
+
+        Important remark:
+        --> For now, both push_to_hub and from_pretrained use segmentation and embedding models from the Hub that are loaded using a config file.
+        In particular, push_to_hub can't be used (for now) to push a custom pipeline with locally modified models.
+
+        Parameters:
+            repo_id (`str`):
+                The name of the repository you want to push your {object} to. It should contain your organization name
+                when pushing to a given organization.
+            embedding_model (`str`):
+                The HF Hub repository containing the segmentation model to be used in the pipeline.
+                If not specified, will use the default pyannote/wespeaker-voxceleb-resnet34-LM model from the Hub.
+            segmentation_model (`str`):
+                The HF Hub repository containing the segmentation model to be used in the pipeline.
+                If not specified, will use the default pyannote/segmentation-3.0 model from the Hub.
+            commit_message (`str`, *optional*):
+                Message to commit while pushing. Will default to `"Upload {object}"`.
+            private (`bool`, *optional*):
+                Whether or not the repository created should be private.
+            token (`bool` or `str`, *optional*):
+                The token to use as HTTP bearer authorization for remote files. If `True`, will use the token generated
+                when running `huggingface-cli login` (stored in `~/.huggingface`). Will default to `True` if `repo_url`
+                is not specified.
+            create_pr (`bool`, *optional*, defaults to `False`):
+                Whether or not to create a PR with the uploaded files or directly commit.
+            revision (`str`, *optional*):
+                Branch to push the uploaded files to.
+            commit_description (`str`, *optional*):
+                The description of the commit that will be created
+            tags (`List[str]`, *optional*):
+                List of tags to push on the Hub.
+            cache_dir: Path or str, optional
+                Path to config cache directory. Defauorch/pyannote" when unset.
+            config_yaml_path (`str`):
+                Path to the original "pyannote/speaker-diarization-3.1" Hub repository containing the default config file to use
+                to create custom speaker diarization config files.
+        """
+
+        api = HfApi()
+
+        _ = api.create_repo(
+            repo_id, private=private, token=token, exist_ok=True, repo_type="model"
+        )
+
+        # Load the pyannote/speaker-diarization-3.1 pipeline config:
+        try:
+            config_yml = hf_hub_download(
```

Comment:
Can this config not be created from the embedding and segmentation models defined in the pipeline? Could you see how it was defined for the original pipeline, and copy the same logic here? Loading the 3.1 pipeline config is a bit brittle, as it won't be compatible with newer versions of the pipeline, which is going to make backwards compatibility difficult

Reply:
I don't think the `config.yaml` file present on the Hub was defined using a pyannote's script (or I haven't find it yet!). It might have been defined manually also.  

I agree that it would be better, I'll check if it's possible to generate it automatically from the pipeline!